lib/spack/spack/package.py

```diff
@@ -2020,7 +2020,7 @@ def all_urls(self):
                 urls.append(args['url'])
         return urls
 
-    def fetch_remote_versions(self):
+    def fetch_remote_versions(self, concurrency=128):
```

Comment:
This magic number works well on my development box, but I wonder if we want to use different defaults for different contexts (e.g. `128` for cli argument and a much lower value for the API)

Reply:
That's what I did last time for:
```console
$ time spack versions -c X  boost
```
on my Lenovo Carbon laptop. Repeating the experiment today, the results changed a bit and 128 is not anymore the clear winner:

Threads | Elapsed time
------------|------------------
16 | 124 sec.
32 | 69 sec.
64 | 56 sec.
128 | 53 sec.

I expect the time it takes for this command to run would be highly inconsistent and to depend on the current latency / flakiness of the network. For instance re-doing the run with 64 threads a few times I got an "outlier" at 93 sec. That said, I can change the default to 32. Let me know what you prefer here.

Reply:
Changed default to 32 in abd85f9